packages/core/content-manager/admin/src/utils/validation.ts

```diff
@@ -327,14 +344,35 @@ const extractValuesFromYupError = (
   };
 };
 
+const isMessageDescriptor = (obj: unknown): obj is MessageDescriptor => {
+  return (
+    obj !== null &&
+    typeof obj === 'object' &&
+    'id' in obj &&
+    typeof obj.id === 'string' &&
+    'defaultMessage' in obj &&
+    typeof obj.defaultMessage === 'string'
+  );
+};
+
 const getInnerErrors = (error: yup.ValidationError) =>
   (error?.inner || []).reduce<Record<string, TranslationMessage>>((acc, currentError) => {
     if (currentError.path) {
-      acc[currentError.path.split('[').join('.').split(']').join('')] = {
-        id: currentError.message,
-        defaultMessage: currentError.message,
-        values: extractValuesFromYupError(currentError?.type, currentError?.params),
-      };
+      const key = currentError.path.split('[').join('.').split(']').join('');
+      const message = currentError.message;
+
+      if (isMessageDescriptor(message)) {
+        acc[key] = {
+          ...(message as MessageDescriptor),
+          values: extractValuesFromYupError(currentError?.type, currentError?.params),
+        };
+      } else {
+        acc[key] = {
+          id: message,
+          defaultMessage: message,
+          values: extractValuesFromYupError(currentError?.type, currentError?.params),
+        };
+      }
```

Comment:
Alright i've just realised the confusion around all this, this function works differently to how the `Form` works managing YupValidation Errors. I must have forgotten about this, lets unify them (imo the form way is easier to understand, unless you disagree?):

https://github.com/strapi/strapi/blob/a8d58e7f7f3f8fd9e661983c7b3220878380e373/packages/core/admin/admin/src/components/Form.tsx#L230